src/main/java/com/avispl/symphony/dal/infrastructure/management/qsc/qsyscore/QSYSCoreAggregatorCommunicator.java

```diff
@@ -226,6 +227,11 @@ public void run() {
 	 */
 	private List<AggregatedDevice> aggregatedDeviceList = Collections.synchronizedList(new ArrayList<>());
 
+	/**
+	 * Keep track all data is collected
+	 */
+	private volatile int isFinishedCollectData = 0;
```

Comment:
The variable name seems a bit confusing. "isSomethingSomething" would normally imply a bool value of sorts, but in this case, it's a counter, so maybe it should be renamed for the clarity? Also please add javadoc description on what the thresholds are